### ISSUE TITLE
|RLlib] DeprecationWarning for custom RLModuleSpec may be too much

### ISSUE THREAD

**Issue by PhilippWillms** [ray-project/ray#45695]
### What happened + What you expected to happen

In nightly built, I detected that `DeprecationWarning` is thrown too easily.

In the reproduction script provided, I make use of an `SingleAgentRLModuleSpec` object relying on my custom RLModule class.
When running PPO Training, the log contains following message:

> 2024-06-03 20:23:59,094 WARNING deprecation.py:50 -- DeprecationWarning: `AlgorithmConfig.rl_module(_enable_rl_module_api=..)` has been deprecated. Use `AlgorithmConfig.api_stack(enable_rl_module_and_learner=..)` instead. This will raise an error in the future!

Checking `algorithm_config.py`, ll. 2994ff, I find out that the warning is always thrown if the _enable_rl_module_api is not explicitly set. A bit more user-friendly solution could be find here before final release of new RLModule API.
```
        if model_config_dict is not NotProvided:
            self._model_config_dict = model_config_dict
        if rl_module_spec is not NotProvided:
            self._rl_module_spec = rl_module_spec

        if _enable_rl_module_api is not NotProvided:
            deprecation_warning(
                old="AlgorithmConfig.rl_module(_enable_rl_module_api=..)",
                new="AlgorithmConfig.api_stack(enable_rl_module_and_learner=..)",
                error=False,
            )
        return self
```

### Versions / Dependencies

ray==3.0.0.dev0

### Reproduction script

          from pprint import pprint
          from typing import OrderedDict
          from typing import Tuple
          
          import gymnasium
          import numpy as np
          import ray
          from gymnasium.spaces import Box
          from gymnasium.spaces import Dict
          from gymnasium.spaces import Discrete
          from gymnasium.wrappers import TransformObservation
          from ray.rllib.algorithms import PPOConfig
          from ray.rllib.core.rl_module.rl_module import SingleAgentRLModuleSpec
          from ray.rllib.env.single_agent_env_runner import SingleAgentEnvRunner
          from ray.rllib.examples.rl_modules.classes.action_masking_rlm import TorchActionMaskRLM
          from ray.tune.registry import register_env
      
      
      class MyRealObsWrapper(TransformObservation):
     
          def __init__(self, env):  # noqa: D107
              super().__init__(env, self.__transform)
      
          def __transform(self, orig_obs):
              new_obs = orig_obs
              for b in new_obs.keys():
                  if b not in ["static_features"]:
                      new_obs[b] = np.reshape(new_obs[b], -1)
              # Important to update the observation space, otherwise the RLlib algorithms will not work
              self.observation_space["observations"] = Box(
                  0, 1, (len(new_obs["observations"]),)
              )
              return new_obs
      
      
      class MyEnv(gymnasium.Env):
          """Custom environment with action masking."""
      
          def __init__(self, *args, **kwargs):  # noqa: D107
              self.action_space = Discrete(3)
              self.observation_space = Dict(
                  {
                      "action_mask": Box(
                          low=0, high=1, shape=(self.action_space.n,), dtype=np.int8
                      ),
                      "observations": Box(
                          low=0.0,
                          high=1.0,
                          shape=(3, 4),
                          dtype=np.float32,
                      ),
                  }
              )
              self.episode_done = False
              self._action_max_helper = np.ones(self.action_space.n, dtype=np.int8)
              self.state = np.zeros((3, 4), dtype=np.float32)
      
          def step(
              self, action: int
          ) -> Tuple[OrderedDict, float, bool, bool, dict]:  # noqa: D102
              # Error handling for invalid action
              if (action < 0) | (action > self.action_space.n):
                  e_string = f"Action [{action}] is not valid! Size of the action space: [{self.action_space.n}]."
                  raise ValueError(e_string)
              if self._action_max_helper[action] == 0:
                  e_string = f"Action [{action}] is not valid as chosen already in episode !"
                  raise ValueError(e_string)      
     
              reward = 0 - action
              self.state[action][0] = 1
              self._action_max_helper[action] = 0
              if all(self._action_max_helper[k] == 0 for k in range(3)):
                  self.episode_done = True
              return self._get_state_repr(), reward, self.episode_done, False, {}
      
          def _get_state_repr(self) -> OrderedDict:
              return {
                  "action_mask": self._action_max_helper,
                  "observations": self.state,
              }
      
          def reset(
              self, *, seed=None, options=None
          ) -> Tuple[OrderedDict, dict]:  # noqa: D102
              self.episode_done = False
              # Initial state representation = shape of the obs space.
              self.state = np.zeros((3, 4), dtype=np.float32)
              # Initial action mask = all actions are allowed.
              self._action_max_helper = np.ones(self.action_space.n, dtype=np.int8)
              return self._get_state_repr(), {}
      
      
      def env_creator(env_config):  # noqa: D103
          env = MyEnv()
          env = MyRealObsWrapper(env)
          return env
      
      
      # Use classic API to register environment
      register_env("myenv_wrapped", env_creator)
      
      if __name__ == "__main__":
          rlm_spec = SingleAgentRLModuleSpec(module_class=TorchActionMaskRLM)
      
          # Algorithm Config, but with the latest RLlib API
          config = (
              PPOConfig()
              .environment("myenv_wrapped")
              .api_stack(
                  enable_rl_module_and_learner=True, enable_env_runner_and_connector_v2=False
              )
              .experimental(_disable_preprocessor_api=True)
              .env_runners(env_runner_cls=SingleAgentEnvRunner, num_env_runners=2)
              .rl_module(rl_module_spec=rlm_spec)
              .framework("torch")
              .resources(num_gpus=1, num_cpus_per_worker=2, num_gpus_per_worker=0.2)
              .training(lr=1e-3, train_batch_size=50, sgd_minibatch_size=10)
          )
      
          algo = config.build()
      
          # run manual training loop and print results after each iteration
          for i in range(10):
              result = algo.train()
              print(f"Training iteration: {i+1} done")
              pprint(result)
      
          ray.shutdown()

### Issue Severity

Low: It annoys or frustrates me.

**Comment by PhilippWillms** [comment]
Checking code base of ray 2.42 release, I see that the code lines mentioned are replaced.


```
 def rl_module(
        self,
        *,
        model_config: Optional[Union[Dict[str, Any], DefaultModelConfig]] = NotProvided,
        rl_module_spec: Optional[RLModuleSpecType] = NotProvided,
        algorithm_config_overrides_per_module: Optional[
            Dict[ModuleID, PartialAlgorithmConfigDict]
        ] = NotProvided,
        # Deprecated arg.
        model_config_dict=DEPRECATED_VALUE,
        _enable_rl_module_api=DEPRECATED_VALUE,
    ) -> "AlgorithmConfig":
...
        if _enable_rl_module_api != DEPRECATED_VALUE:
            deprecation_warning(
                old="AlgorithmConfig.rl_module(_enable_rl_module_api=..)",
                new="AlgorithmConfig.api_stack(enable_rl_module_and_learner=..)",
                error=True,
            )
```
